### PR TITLE
:sparkles: add full support of any type of graphql resolver

### PR DESCRIPTION
**Pull request by Taranys**
Hi,

I create this PR to support all type of resolvers. Right now, array of resolvers or scalar are not working.

I took some time to enable TS types everywhere to be sure that you will be warn after an upgrade if appolo add a new type of resolver.

PS: I also had to force graphql version everywhere in the project
PS2: Impossible to run the project without a default password on docker-compose file

### CODE REVIEW

File: graphql/package.json [hunk 1]
```diff
@@ -29,6 +29,6 @@
     "@cuillere/core": "^0.10.0"
   },
   "devDependencies": {
-    "graphql": "^15.3.0"
+    "graphql": "~14"
```
Comment (nlepage) [inline graphql/package.json:32]:
```suggestion
    "apollo-server-koa": "^2.15.1"
  },
  "dependencies": {
    "graphql": "~14"
  },
```

File: graphql/src/index.ts [hunk 2]
```diff
@@ -1,26 +1,54 @@
...
+function applyToResolvers(fn: FieldResolverWrapper): (resolvers: IResolvers) => IResolvers;
+function applyToResolvers(fn: FieldResolverWrapper): (resolvers: IResolvers[]) => IResolvers[];
+function applyToResolvers(fn: FieldResolverWrapper): (resolvers: OneOrMany<IResolvers>) => OneOrMany<IResolvers> {
+  return (resolvers: IResolvers | IResolvers[]) => {
+    if (isResolverArray(resolvers)) {
+      return resolvers.map(resolver => applyToResolvers(fn)(resolver))
+    }
+
+    // Usage of .reduce() to keep type inference
+    return Object.entries(resolvers).reduce((resolver: IResolvers, [key, value]) => {
```
Comment (nlepage) [inline graphql/src/index.ts:29]: I'd rather have a for loop assigning to a const object.

File: graphql/src/index.ts [hunk 3]
```diff
@@ -1,26 +1,54 @@
...
+}
 
-  const treeToResolversTree = (tree: Record<string, any>) => Object.fromEntries(
-    Object.entries(tree).map(([key, value]) => [
-      key,
-      typeof value === 'function' ? fnToResolver(value) : treeToResolversTree(value),
-    ]),
-  )
+function applyToObject(fn: FieldResolverWrapper): (resolvers: IResolverObject) => IResolverObject {
+  return resolverObject => Object.entries(resolverObject).reduce((resolver: IResolverObject, [key, value]) => {
```
Comment (nlepage) [inline graphql/src/index.ts:41]: Same here

**Review by nlepage — CHANGES_REQUESTED** [review]
Some suggestions

**Review by EmrysMyrddin — APPROVED** [review]

**Review by nlepage — APPROVED** [review]

**Pull request merged**